sql/core/src/main/scala/org/apache/spark/sql/jdbc/DerbyDialect.scala

```diff
@@ -50,4 +50,17 @@ private object DerbyDialect extends JdbcDialect {
   override def renameTable(oldTable: String, newTable: String): String = {
     s"RENAME TABLE $oldTable TO $newTable"
   }
+
+  // Derby currently doesn't support comment on table. Here is the ticket to add the support
+  // https://issues.apache.org/jira/browse/DERBY-7008
+  override def createTable(
+      table: String,
+      strSchema: String,
+      createTableOptions: String,
+      tableComment: String): Array[String] = {
+    if (!tableComment.isEmpty) {
+      logWarning("Cannot create JDBC table comment. The table comment will be ignored.")
```

Comment:
So you are treating `comment`  as a special table property that is OK to ignore. Is `comment` the only one?

Reply:
We don't append comment to the end of the CREATE TABLE statement. It's a separate sql statement. If creating comment fails, it doesn't affect the table creation, so it can be treated as a special property. We append all the other properties to the end of the CREATE TABLE statement as createTableOptions.